python/pyspark/testing/utils.py

```diff
@@ -209,3 +232,105 @@ def check_error(
         self.assertEqual(
             expected, actual, f"Expected message parameters was '{expected}', got '{actual}'"
         )
+
+
+def assertDataFrameEqual(
+    df: DataFrame, expected: Union[DataFrame, List[Row]], ignore_row_order: bool = True
+):
+    """
+    A util function to assert equality between DataFrames `df` and `expected`, with
+    optional arg `ignore_row_order`.
+    For float values, assert approximate equality (1e-5 by default).
```

Comment:
If we're exposing this as an API, it should have documentation that follows NumPy documentations. Let's probably at least have [Parameters](https://numpydoc.readthedocs.io/en/latest/format.html#parameters).